android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.kt

```diff
@@ -0,0 +1,511 @@
+      handler.stopTrackingPointer(pointerId)
+    }
+  }
+
+  /**
+   * isViewAttachedUnderWrapper checks whether all of parents for view related to handler
+   * view are attached. Since there might be an issue rarely observed when view
+   * has been detached and handler's state hasn't been change to canceled, failed or
+   * ended yet. Probably it's a result of some race condition and stopping delivering
+   * for this handler and changing its state to failed of end appear to be good enough solution.
+   */
+  private fun isViewAttachedUnderWrapper(view: View?): Boolean {
+    if (view == null) {
+      return false
+    }
+    if (view === wrapperView) {
+      return true
+    }
+    var parent = view.parent
+    while (parent != null && parent !== wrapperView) {
+      parent = parent.parent
+    }
+    return parent === wrapperView
+  }
+
+  private fun extractCoordsForView(view: View?, event: MotionEvent, outputCoords: FloatArray) {
+    if (view === wrapperView) {
+      outputCoords[0] = event.x
+      outputCoords[1] = event.y
+      return
+    }
+    require(!(view == null || view.parent !is ViewGroup)) { "Parent is null? View is no longer in the tree" }
+    val parent = view.parent as ViewGroup
+    extractCoordsForView(parent, event, outputCoords)
+    val childPoint = tempPoint
+    transformTouchPointToViewCoords(outputCoords[0], outputCoords[1], parent, view, childPoint)
+    outputCoords[0] = childPoint.x
+    outputCoords[1] = childPoint.y
+  }
+
+  private fun addAwaitingHandler(handler: GestureHandler<*>) {
+    for (i in 0 until awaitingHandlersCount) {
+      if (awaitingHandlers[i] === handler) {
+        return
+      }
+    }
+    check(awaitingHandlersCount < awaitingHandlers.size) { "Too many recognizers" }
+    awaitingHandlers[awaitingHandlersCount++] = handler
+    handler.mIsAwaiting = true
+    handler.mActivationIndex = activationIndex++
+  }
+
+  private fun recordHandlerIfNotPresent(handler: GestureHandler<*>, view: View) {
+    for (i in 0 until gestureHandlersCount) {
+      if (gestureHandlers[i] === handler) {
+        return
+      }
+    }
+    check(gestureHandlersCount < gestureHandlers.size) { "Too many recognizers" }
+    gestureHandlers[gestureHandlersCount++] = handler
+    handler.mIsActive = false
+    handler.mIsAwaiting = false
+    handler.mActivationIndex = Int.MAX_VALUE
+    handler.prepare(view, this)
+  }
+
+  private fun recordViewHandlersForPointer(view: View, coords: FloatArray, pointerId: Int): Boolean {
+    val handlers = handlerRegistry.getHandlersForView(view)
+    var found = false
+    handlers?.let {
+      var i = 0
+      val size = it.size
+      while (i < size) {
+        val handler = it[i]
+        if (handler.isEnabled && handler.isWithinBounds(view, coords[0], coords[1])) {
+          recordHandlerIfNotPresent(handler, view)
+          handler.startTrackingPointer(pointerId)
+          found = true
+        }
+        i++
+      }
+    }
+    return found
```

Comment:
I'm not sure whether this change makes it possible for `ConcurrentModificationException` to occur or not (when attaching a new handler to a view). If that's the case, I think we can change it to use the range operator.
```suggestion
    var found = false
    handlerRegistry.getHandlersForView(view)?.let { handlers ->
      for (handler in handlers) {
        if (handler.isEnabled && handler.isWithinBounds(view, coords[0], coords[1])) {
          recordHandlerIfNotPresent(handler, view)
          handler.startTrackingPointer(pointerId)
          found = true
        }
      }
    }
    return found
```

Reply:
I went with the safe way here by using ranges.
Fixed in c4c41eb, 52b833d.